bundles/io/org.eclipse.smarthome.io.transport.mdns/src/main/java/org/eclipse/smarthome/io/transport/mdns/internal/MDNSClientImpl.java

```diff
@@ -86,10 +137,21 @@
     }
 
     @Activate
-    public void activate() {
+    protected void activate() {
+        start();
+    }
+
+    private void start() {
         for (InetAddress address : getAllInetAddresses()) {
             createJmDNSByAddress(address);
         }
+        for (ServiceDescription description : activeServices) {
```

Comment:
Your assumption is wrong, because registerService is a public method and not called only here. But you're completely right, that there is a wrong behaviour: the service must of course not be added again in this loop. I fixed this.

Reply:
My assumption said: "resp. the registerService method had already been called for the service in question".
I don't see any change on this yet...

Reply:
Did you see my last comment here? Am I missing something?

Reply:
The start method is used in initialization, there are no active services at that moment.
And it is used when network configuration changes. In that case all active MDNS instances are terminated and the services have to be reregistered on the newly created MDNS instances.
I can see no problem here.

Reply:
> The start method is used in initialization, there are no active services at that moment.

So you agree that this loop is useless in that case? Ok. (Might then be better not even be executed at all.)

> And it is used when network configuration changes. In that case all active MDNS instances are terminated 

... by calling `deactivate`, which should actually clear `activeServices`, which is does not (yet) do (I thought it did, but I noticed that this is so far only done in `unregisterAllServices`).
I think the `onChanged` should more explicitly contain the code that it is supposed to execute - `deactivate` is definitely not a good method to call there (at least if not followed by an `activate`...).